src/runtime_src/core/tools/common/tests/TestAIEReconfigOverhead.cpp

```diff
@@ -0,0 +1,182 @@
+  xrt::bo bo_ofm(working_dev, buffer_size, XRT_BO_FLAGS_HOST_ONLY, kernel.group_id(argno++));
+  xrt::bo bo_inter(working_dev, inter_size, XRT_BO_FLAGS_HOST_ONLY, kernel.group_id(argno++));
+  xrt::bo bo_instr(working_dev, instr_size*sizeof(int), XCL_BO_FLAGS_CACHEABLE, kernel.group_id(argno));
+  xrt::bo bo_instr_no_op(working_dev, instr_size*sizeof(int), XCL_BO_FLAGS_CACHEABLE, kernel.group_id(argno++));
+  argno++;
+  xrt::bo bo_mc(working_dev, 16, XRT_BO_FLAGS_HOST_ONLY, kernel.group_id(argno++));
+
+  init_instr_buf(bo_instr, dpu_instr);
+  //Create ctrlcode with NOPs
+  std::memset(bo_instr_no_op.map<char*>(), 0, instr_size);
+
+  // map input buffer
+  // Incremental byte pattern
+  auto ifm_mapped = bo_ifm.map<int*>();
+  for (size_t i = 0; i < word_count; i++)
+    ifm_mapped[i] = (int)(i % word_count);
+
+  //Sync BOs
+  bo_instr.sync(XCL_BO_SYNC_BO_TO_DEVICE);
+  bo_mc.sync(XCL_BO_SYNC_BO_TO_DEVICE);
+  bo_ifm.sync(XCL_BO_SYNC_BO_TO_DEVICE);
+
+  //Log
+  if(XBUtilities::getVerbose()) { 
+    logger(ptree, "Details", boost::str(boost::format("Buffer size: '%f'MB") % buffer_size_mb));
+    logger(ptree, "Details", boost::str(boost::format("No. of iterations: '%f'") % itr_count));
+  }
+
+  auto start = std::chrono::high_resolution_clock::now();
+  for (int i = 0 ;i < itr_count ; i++){
+    try{
+      auto run = kernel(1, bo_ifm, NULL, bo_ofm, bo_inter, bo_instr_no_op, instr_size, bo_mc);
+      run.wait2();
+    }
+    catch (const std::exception& ex)
+    {
+      logger(ptree, "Error", ex.what());
+      ptree.put("status", test_token_failed);
+      return ptree;
+    }
+    bo_ofm.sync(XCL_BO_SYNC_BO_FROM_DEVICE);
+  }
+  auto end = std::chrono::high_resolution_clock::now();
+  float elapsedSecsNoOpAverage = std::chrono::duration_cast<std::chrono::duration<float>>(end-start).count();
+  elapsedSecsNoOpAverage /= itr_count;
+
+  start = std::chrono::high_resolution_clock::now();
+  for (int i = 0; i< itr_count; i++)
+  {
+    try{
+      auto run = kernel(1, bo_ifm, NULL, bo_ofm, bo_inter, bo_instr, instr_size, bo_mc);
+      run.wait2();
+    }
+    catch (const std::exception& ex)
+    {
+      logger(ptree, "Error", ex.what());
+      ptree.put("status", test_token_failed);
+      return ptree;
+    }
+    bo_ofm.sync(XCL_BO_SYNC_BO_FROM_DEVICE);
+    auto *ofm_mapped = bo_ofm.map<int8_t*>();
+    if(std::memcmp(ifm_mapped, ofm_mapped + StartAddr, word_count)){
+      logger(ptree, "Error", "Value read back does not match reference for array reconfiguration instruction buffer");
+      ptree.put("status", test_token_failed);
+      return ptree;
+    }
+  }
+
+  end = std::chrono::high_resolution_clock::now();
+  float elapsedSecsAverage = std::chrono::duration_cast<std::chrono::duration<float>>(end-start).count();
+  elapsedSecsAverage /= itr_count;
+  float overhead = elapsedSecsAverage - elapsedSecsNoOpAverage; 
+  logger(ptree, "Details", boost::str(boost::format("Array reconfiguration overhead: '%.7f's") % (overhead)));
```

Comment:
Let's convert s to ms so it shows up as "3.4 ms" which is more readable and limit the decimal place to 1 for consistency with other tests